source/MaterialXRender/ImageHandler.cpp

```diff
@@ -184,6 +161,53 @@ void ImageHandler::releaseRenderResources(ImagePtr)
 {
 }
 
+ImagePtr ImageHandler::loadImage(const FilePath& filePath)
+{
+    string extension = stringToLower(filePath.getExtension());
+    for (ImageLoaderPtr loader : _imageLoaders[extension])
+    {
+        ImagePtr image;
+        try
+        {
+            image = loader->loadImage(filePath);
+        }
+        catch (std::exception& e)
+        {
+            std::cerr << "Exception in image I/O library: " << e.what() << std::endl;
```

Comment:
For a side discussion: would like to resolve how to route error messages to places other than stderr. I think we did something for texture baker which allowed you to set the output stream which would allow routing to a stringstream.

Reply:
That's a good point, and I like the idea of extending the TextureBaker pattern to ImageHandler in the future, where the output stream can be modified by the client.